A11Y Setup

hamo225 opened the pull request:
RISDEV-6515

hamo225 committed:
chore: replace axe-playwright with axe-core

- Uninstalled axe-playwright and installed axe-core
- Added a11y utilities for tests

RISDEV-6515

hamo225 committed:
test(a11y): add accessibility test for Amending Laws page

- Scans for accessibility violations using axe-core

RISDEV-6515

hamo225 committed:
fix(a11y): resolve initial accessibility violations

RISDEV-6515

hamo225 committed:
chore: update Playwright and Vitest config for a11y tests

- Updated `playwright.config.ts` to include a11y project
- Ensured `vitest.config.ts` excludes Playwright tests

RISDEV-6515

hamo225 committed:
ci: integrate a11y tests into pipeline

RISDEV-6515

hamo225 committed:
docs: add accessibility testing instructions to README

RISDEV-6515

hamo225 committed:
ci: ensure a11y tests run after authentication setup

RISDEV-6515

andreasphil commented on .github/workflows/end-to-end-tests.yml:
```diff
@@ -58,7 +58,11 @@ jobs:
           npm run preview -- --port 5173 &
           DEV_PID=$!
-          npm run test:e2e -- --project ${{ inputs.project }} --repeat-each 1
+          if [[ "${{ inputs.project }}" == "a11y" ]]; then
```
Question: why is this necessary? This should work:

```sh
npm run test:e2e -- --project a11y --repeat-each 1
```

As far as I can tell both branches of the `if` to the same thing.

andreasphil commented on frontend/README.md:
```diff
@@ -66,6 +66,18 @@ npm run test:e2e -- --project msedge --repeat-each 1
+
+The project includes automated accessibility (a11y) testing using [axe-core](https://github.com/dequelabs/axe-core).
+
+### **Running Accessibility Tests**
```
nitpick: remove `**`

andreasphil commented on frontend/a11y/a11y.utils.ts:
```diff
@@ -0,0 +1,20 @@
+import { AxeBuilder } from "@axe-core/playwright"
```
todo: rename this file to `a11y-utils.ts` or something like that. `.utils.ts` (with a dot) suggests a specific convention for filenames or purpose of the file which we don't have for utils.

andreasphil reviewed (changes requested):
@hamo225 what about tests for the other views? I think we discussed this but don't remember what the agreement was.

hamo225 committed:
ci: resolve technical review comments on a11y test execution

- includes renaming of the utils file and small change to the readme

RISDEV-0000

hamo225 commented:
@andreasphil We agreed one view for now, since we were not sure what views will stay
